### PR TITLE
Redoing PR for fixes #57 and #60 (and maybe #59)

### LINKED ISSUE #57: Sequencer run fails
Input available here:  http://45.55.201.87:8000/data/7801ba1e-b688-41ee-85c5-b0af9f2edc4f/input.zip

Can sequence id's be floats?  

Traceback (most recent call last):
  File "/home/mr/modelrunner/models/mvmax_sequencer.py", line 36, in <module>
    results = model.sequence()
  File "/home/mr/miniconda/envs/sequencer/lib/python2.7/site-packages/sequencer/Models.py", line 13, in sequence
    super(EnergyMaximizeReturn, self).sequence()
  File "/home/mr/miniconda/envs/sequencer/lib/python2.7/site-packages/sequencer/Sequencer.py", line 139, in sequence
    self._build_edge_wkt()
  File "/home/mr/miniconda/envs/sequencer/lib/python2.7/site-packages/sequencer/Sequencer.py", line 241, in _build_edge_wkt
    self.networkplan.network.edge[fnode][tnode]['distance'] = float(self.networkplan._distance(fnode, tnode))
  File "/home/mr/miniconda/envs/sequencer/lib/python2.7/site-packages/sequencer/NetworkPlan.py", line 155, in _distance
    self.coord_values[first_index], self.coord_values[second_index]
TypeError: list indices must be integers, not numpy.float64

### LINKED ISSUE #60: add test with fake nodes
need a network where some nodes don't have corresponding demand nodes for testing that case

### PR DESCRIPTION
**Pull request by chrisnatali**
Closes #60, #59 
Assigning to @vr2262 (as my trigger was too quick on merging my own PR last time)

### CODE REVIEW

**Comment by vr2262** [comment]
I'll take care of this tomorrow morning.

File: sequencer/NetworkPlan.py [hunk 1]
```diff
@@ -255,11 +280,15 @@ def network(self):
         return self._network
     
     @property
+    def original_metrics(self):
+        """returns the original (unprocessed) metrics data_frame"""
+        return self._original_metrics
```
Comment (vr2262) [inline sequencer/NetworkPlan.py]: Why is `original_metrics` a property and not just an attribute?

File: sequencer/NetworkPlan.py [hunk 2]
```diff
@@ -19,31 +17,32 @@ class NetworkPlan(object):
...
-    NetworkPlan('/Users/blogle/Downloads/1643/networks-proposed.shp', 
-                '/Users/blogle/Downloads/1643/metrics-local.csv')
     """
     TOL = .5 # meters at the equator, tolerance is stricter towards the poles
 
-    def __init__(self, shp, csv, **kwargs):
-        self.shp_p, self.csv_p = shp, csv
+    def __init__(self, network, metrics, **kwargs):
         self.priority_metric = kwargs['prioritize'] if 'prioritize' in kwargs else 'population'
+        self.proj = kwargs['proj'] if 'proj' in kwargs else 'utm'
```
Comment (vr2262) [inline sequencer/NetworkPlan.py]: This would be better as

```
self.proj = kwargs.get('proj', 'utm')
```

File: sequencer/Sequencer.py [hunk 3]
```diff
@@ -235,7 +234,9 @@ def _build_edge_wkt(self):
         r = self.results
         # Iterate through the nodes and their parent
         for rank, fnode, tnode in zip(r.index, r['Sequence..Upstream.id'], r['Sequence..Vertex.id']):
-            if not np.isnan(fnode):
+            if fnode is not None:
+                if np.any(np.mod([fnode, tnode], 1) != 0):
+                    raise Exception('Non-integral node index in results.')
```
Comment (vr2262) [inline sequencer/Sequencer.py]: Do you want to keep this check? It doesn't do anything anymore.

**Comment by vr2262** [comment]
@chrisnatali 

Tasks before merge:
- [ ] https://github.com/SEL-Columbia/Sequencer/pull/62#discussion-diff-61439140
- [ ] https://github.com/SEL-Columbia/Sequencer/pull/62#discussion-diff-61441162
- [ ] https://github.com/SEL-Columbia/Sequencer/pull/62#discussion_r61443860

File: sequencer/NetworkPlan.py (hunk 1 above)
Comment (chrisnatali) [inline sequencer/NetworkPlan.py]: to keep consistent with _network and _metrics.  Better ideas welcome.  
I don't like original_metrics being there at all, but the dependency in sequencer was hard to get rid of quickly.  Will address that with large refactoring or rewrite.

File: sequencer/NetworkPlan.py (hunk 2 above)
Comment (chrisnatali) [inline sequencer/NetworkPlan.py]: Cool

File: sequencer/NetworkPlan.py (hunk 1 above)
Comment (vr2262) [inline sequencer/NetworkPlan.py]: Oh yeah, now that you mention it there's a lot of property weirdness going on here... May as well clean it up in a different PR.

File: sequencer/Sequencer.py (hunk 3 above)
Comment (chrisnatali) [inline sequencer/Sequencer.py]: Will get rid of...it obscures things now

File: sequencer/NetworkPlan.py (hunk 1 above)
Comment (vr2262) [inline sequencer/NetworkPlan.py]: Opened #63

File: sequencer/Tests/Test_Suite.py [hunk 4]
```diff
@@ -46,6 +46,64 @@ def gen_data():
...
+    """
+    Make sure we work with fake nodes
+    """
+    
+    # for now, just make sure it runs without exceptions
+    metrics, network = gen_data_with_fakes()
+    nwp = NetworkPlan(network, metrics, prioritize='Population', proj='wgs4')
+    model = EnergyMaximizeReturn(nwp)
+    model.sequence()
+    #todo:  check the result
```
Comment (chrisnatali) [inline sequencer/Tests/Test_Suite.py]: Though this catches fix#57, we should make sure the results are valid in some way

Comment (vr2262) [inline sequencer/Tests/Test_Suite.py]: Could you check against a known-good result?

**Comment by vr2262** [comment]
Is this a good and happy PR now?

**Comment by chrisnatali** [comment]
I think so.  Let's keep in mind #64 (this basically needs an overhaul, so lots of nit-picking isn't productive).  If you don't see anything glaring, I'm gonna do a test on some of the failed scenarios in modelrunner and then release.

**Comment by vr2262** [comment]
Ship it :shipit:

**Pull request merged**